sklearn/neighbors/_base.py

```diff
@@ -276,15 +276,22 @@ def _radius_neighbors_from_graph(graph, radius, return_distance):
     indices = indices.astype(np.intp, copy=no_filter_needed)
 
     if return_distance:
-        neigh_dist = np.array(np.split(data, indptr[1:-1]), dtype=object)
-    neigh_ind = np.array(np.split(indices, indptr[1:-1]), dtype=object)
+        neigh_dist = _array_of_objects(np.split(data, indptr[1:-1]))
+    neigh_ind = _array_of_objects(np.split(indices, indptr[1:-1]))
 
     if return_distance:
         return neigh_dist, neigh_ind
     else:
         return neigh_ind
 
 
+def _array_of_objects(sequence):
```

Comment:
Define the function locally will be enough since we don't use elsewhere.

Reply:
We have certainly used this idiom elsewhere in scikit-learn. I don't mind refactoring it.

Reply:
I don't think so because we should have get the same warning regarding the `dtype=object` raised by numpy, you don't think so?